File: content/php-library/current/source/includes/connect/network-compression.php
<?php

// start-enable-client
$client = new MongoDB\Client(
    'mongodb://<hostname>:<port>',
    ['compressors' => 'snappy,zstd,zlib'],
);
// end-enable-client

// start-enable-uri
$uri = 'mongodb://<hostname>:<port>/?compressors=snappy,zstd,zlib';
$client = new MongoDB\Client($uri);
// end-enable-uri

// start-set-level-client
$uriOptions = [
    'compressors'  => 'zlib',
    'zlibCompressionLevel' => 1,
];

$client = new MongoDB\Client(
    'mongodb://<hostname>:<port>',
    $uriOptions,
);
// end-set-level-client

// start-set-level-uri
$uri = 'mongodb://<hostname>:<port>/?compressors=zlib&zlibCompressionLevel=1';
$client = new MongoDB\Client($uri);
// end-set-level-uri

File: content/ruby-driver/upcoming/source/includes/connect/network-compression.rb
# start-connection-string
uri = "mongodb://<hostname>:<port>/?compressors=zlib,snappy"
client = Mongo::Client.new(uri)
# end-connection-string

# start-client-settings
client = Mongo::Client.new(["<hostname>:<port>"],
    compressors: ["zlib", "snappy"])
# end-client-settings